workspaces/ui-v2/src/hooks/useCapturesHook.tsx

```diff
@@ -23,13 +23,18 @@ export function useCaptures(): {
       shouldCancel = true;
     };
   }, [capturesService]);
-  if (captures === null) {
-    return { captures: [], loading: true };
-  }
-  return {
-    captures,
-    loading: false,
-  };
+
+  const returnValue = useMemo(() => {
+    if (captures === null) {
+      return { captures: [], loading: true };
+    }
+    return {
+      captures,
+      loading: false,
+    };
+  }, [captures]);
+
+  return returnValue;
```

Comment:
@niclim what do you think of this? I was depending on the return value of `useCaptures` in another hook, and it was infinite-looping I _think_ because this function was returning a _new_ object every time, which wasn't referentially equal to the old one, and React doesn't try to do deep equality checks on dependencies. 

This did fix the infinite looping issue... but it's not necessarily the prettiest. Thoughts?

Reply:
You could move all this state into `captures` - see similar patterns we have with `AsyncStatus` and other loading states. https://github.com/opticdev/optic/blob/develop/workspaces/ui-v2/src/hooks/useInteraction.tsx#L13-L34

You are correct in why it infinite loops - moving the state into useState will prevent this (so we aren't creating a new object on return)

We'll probably also want to move this into redux as well for a single source of truth, but we can do that in a separate PR